trademapper/js/trademapper.videoexport.js

```diff
@@ -0,0 +1,116 @@
+define(["gif", "jquery", "util"], function (GIF, $, util) {
+
+	return {
+		// button: button which when clicked initiates the export
+		// trademapper: trademapper instance
+		init: function (button, trademapper) {
+			var self = this;
+
+			this.trademapper = trademapper;
+
+			// provide a way to trigger events, even though this component has no
+			// DOM element of its own
+			this.eventFirer = $({});
+
+			button.on("click", function () {
+				self.eventFirer.trigger("start");
```

Comment:
Elsewhere I think I've seen you use the 'bind' approach to associating the 'owning' object with the event handler. Here you use the 'self' approach. I think we should stick to one or the other. Looking at the code it seems the 'bind' approach is used more often (greping the code for 'self' returns only one instance, in loadTopoJsonFromURL, where it's not actually used)